core/src/main/java/org/bitcoinj/core/ECKey.java

```diff
@@ -574,31 +575,49 @@ public ECDSASignature toCanonicalised() {
             }
         }
 
-        public static ECDSASignature decodeFromDER(byte[] bytes) throws IllegalArgumentException {
-            ASN1InputStream decoder = null;
-            try {
-                decoder = new ASN1InputStream(bytes);
-                DLSequence seq = (DLSequence) decoder.readObject();
-                if (seq == null)
-                    throw new IllegalArgumentException("Reached past end of ASN.1 stream.");
-                ASN1Integer r, s;
-                try {
-                    r = (ASN1Integer) seq.getObjectAt(0);
-                    s = (ASN1Integer) seq.getObjectAt(1);
-                } catch (ClassCastException e) {
-                    throw new IllegalArgumentException(e);
+        private static int readLength(ByteArrayInputStream input) {
+            int len = input.read();
+            if ((len & 0x80) == 0) {
+                return len;
+            } else {
+                int n = len - 0x80;
+                int len1 = 0;
+
+                while (n > 0) {
+                    len1 = (len1 << 8) + input.read();
+                    n = n - 1;
                 }
-                // OpenSSL deviates from the DER spec by interpreting these values as unsigned, though they should not be
-                // Thus, we always use the positive versions. See: http://r6.ca/blog/20111119T211504Z.html
-                return new ECDSASignature(r.getPositiveValue(), s.getPositiveValue());
+
+                return len1;
+            }
+        }
+
+        private static ECDSASignature decodeSignatureLax(ByteArrayInputStream input) {
```

Comment:
Well, after some research I probably do know: we get `r` and `s` from `ASN1InputStream` as instance of `ASN1Integer` and `1.56` release notes specifically say that "Strict encoding enforcement has been introduced for ASN1Integer" which is probably incompatible with "lax DER signatures" which were allowed in old Bitcoin transactions. Should I put all this into description?